### ISSUE TITLE
feat(ml): treat the backbone's inherited scale invariance as an experiment axis

### ISSUE THREAD

**Issue by LukeSantossz** [LukeSantossz/visiosoil-app#204]
## Description

ImageNet pretraining is trained to be invariant to scale, and soil texture classification is entirely about absolute particle size. The backbone therefore arrives having learned to discard the one signal this task needs, and how much of it to retrain is an experimental axis rather than a default. Add that axis to E0, and add a third option to it: pretraining a soil-specific backbone on sample identity, which needs no texture label.

## Context

Raised on 2026-09-01 from the transfer-learning literature in the project owner's reference notes, and it is a principle the project has not applied to itself:

> The invariances of the pretraining are inherited along with the features. If the new task depends on precisely a distinction the pretraining learned to ignore, the pretrained model is actively worse than random initialisation on that axis. It is worth inventorying which invariances the pretraining built in before assuming that transferring helps.

The textbook example is horizontal mirroring and traffic signs: a network built on ImageNet features confuses *turn left* with *turn right*, because mirroring is one of ImageNet's standard augmentations and the two classes become the same picture.

**The axis that bites here is scale, and it bites harder.** The standard ImageNet training recipe uses random resized crop — scale jitter far more aggressive than a mirror — so the resulting features are deliberately scale-invariant. A cat is a cat at any apparent size. Meanwhile the Embrapa textural groups are defined by particle size distribution, and [ADR 0018](../blob/main/docs/adr/0018-model-sees-fixed-size-greyscale-patches-and-their-spread-is-a-quality-signal.md) has the model looking at patches of fixed **physical** size for exactly that reason. The backbone was optimised to throw away the axis the label lives on.

Three consequences, and none of them is currently written down anywhere:

1. **[SPEC 0037](../blob/main/docs/specs/0037-scale-normalised-greyscale-patch-pipeline.md)'s canonical millimetres-per-pixel normalisation is a precondition for transfer, not tidiness.** The backbone will not recover scale from the image, because it was trained not to. Handing it a scale-normalised patch is what makes the pretrained features usable at all.
2. **Freezing the stem is the worst option here, not the safest one.** Scale invariance is built in the early convolutional layers, which is exactly what `model.freeze_backbone: true` preserves. The usual intuition — freeze more when data is scarce — points the wrong way on this axis.
3. **Mirroring and 90° rotation are free.** Soil in a dish has no canonical orientation, so ImageNet's flip invariance costs nothing here, unlike the traffic-sign case. This corroborates #195's symmetry argument from the transfer side.

**The third arm, and why it fits this dataset.** Contrastive pretraining builds representations from pairs, and the usual recipe takes positives as two augmentations of one image — which teaches invariance to whatever was used as the augmentation, and would reintroduce scale invariance if crop or zoom were in that set. A different pairing avoids it entirely: **positives are two patches of the same physical sample, negatives are patches of other samples.** That uses `sample_id`, which SPEC 0040 now declares for every row, needs no texture label at all, and can consume all 221 photographs including Siltosa and the transported population. The invariances it teaches are position within the dish and illumination; scale stays a variance, not an invariance.

This matters because of what #181 measured: 105 sample groups is very little labelled data, but sample identity is not scarce. It is a way to extract far more signal per photograph than four class labels do.

## Current State

`ml/config.yaml`:

```yaml
model:
  architecture: "mobilenetv2"
  freeze_backbone: true
  dropout: 0.5
  unfreeze_at_epoch: 10
  unfreeze_layers: 50
```

`ml/src/model.py` builds MobileNetV2 with ImageNet weights and `unfreeze_model` releases the last `unfreeze_layers` for phase two. `unfreeze_layers: 50` of MobileNetV2's ~155 layers means **the stem is never trained**, which under the argument above is the configuration least able to recover the discarded axis. No record justifies 50; it is a default.

#179 is adjacent and not the same: it reports that phase-two fine-tuning overwrites the ImageNet BatchNorm statistics, which changes how much of the backbone is really being adapted. Whatever this issue concludes, #179's fix has to land first or the arms are not comparable.

#183 re-scopes E0 to four arms — colour histogram, classical texture descriptors, fine-tuned CNN, label-shuffled control. This issue does not add a fifth arm; it says the third arm is **not one configuration** and names the axis it varies along.

## Proposed Solution

1. **Make the fine-tuning depth an explicit E0 axis, not a config default.** At minimum: stem frozen (the status quo), and the whole backbone fine-tuned. The comparison is diagnostic in itself — if the frozen stem wins at this data size, the scale-invariance argument is wrong or is dominated by overfitting, and that is worth knowing before any further investment.
2. **Add a sample-identity contrastive pretraining arm.** Positives are patches sharing a `sample_id`; negatives are patches of other samples; the augmentation set must exclude every scale and crop transform, and that exclusion is the point rather than an implementation detail. The resulting backbone then replaces the ImageNet one in the fine-tuning arms.
3. **Record the inventory of inherited invariances** in the architecture study, naming which of ImageNet's augmentations are harmless here (mirroring, rotation) and which are adversarial (scale, and arguably colour jitter, since soil hue is part of the signal).
4. **Amend SPEC 0037 or its successor** with consequence 1 above — that the canonical normalisation is what makes a pretrained backbone usable, not only what makes the training set self-consistent.

Criteria for this shape: the axis is cheap to test because the arms share every other component; the contrastive arm is the only proposal on the table that increases the signal extracted per photograph rather than the number of tensors; and all three are decided by the same E0 test set, so they cost no additional evaluation machinery.

**What would make this unnecessary.** If E0's classical texture descriptor arm beats the CNN, per #183, the deep pipeline is not earning its runtime and none of this matters. That arm should be read first.

## Acceptance Criteria

- [ ] The architecture study records which ImageNet pretraining invariances are harmless for this task and which are adversarial, with the scale argument stated in full.
- [ ] Fine-tuning depth is an E0 axis with at least two settings, and `model.unfreeze_layers` carries the reasoning behind its value or is marked provisional against this issue.
- [ ] A sample-identity contrastive pretraining arm is specified: pairing rule, the augmentation set and the explicit exclusion of every scale and crop transform, and the loss.
- [ ] The contrastive arm is shown to use `sample_id` from the manifest and no texture label, so it can consume Siltosa and the train-only population.
- [ ] Every arm reports on the same evaluation set as #183's other arms, so the comparison is paired.
- [ ] #179 lands before any arm is run, and the acceptance record says so.
- [ ] SPEC 0037's normalisation is documented as a precondition for transfer, not only for training-set consistency.
